flutter/news/lib/dependencies_injection.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:convert';
+
+import 'package:get_it/get_it.dart';
+import 'package:news/features/frontpage/data/datasource/articles_local_data_source.dart';
+import 'package:news/features/frontpage/data/datasource/articles_remote_data_source.dart';
+import 'package:news/features/frontpage/data/repositories/articles_repository.dart';
+import 'package:news/features/frontpage/presentation/bloc/articles_cubit.dart';
+
+import 'core/http_client.dart';
+import 'core/news_api_client.dart';
+
+final getIt = GetIt.instance;
+
+Future<void> init() async {
+  getIt.registerFactory(() => ArticlesCubit(repository: getIt()));
+
+  getIt.registerLazySingleton(
+    () =>
+        ArticlesRepository(localDataSource: getIt(), remoteDataSource: getIt()),
+  );
+
+  getIt.registerLazySingleton(() => ArticlesLocalDataSource(getIt()));
+
+  getIt.registerLazySingleton(() => ArticlesRemoteDataSource(client: getIt()));
+
+  getIt.registerLazySingleton(() => const JsonCodec());
+
+  getIt.registerLazySingleton(() => NewsApiClient(getIt()));
+
+  getIt.registerLazySingleton(() => createDio());
```

Comment:
I would like you to reflect about something.

`createDio` creates a DIO having on the properties the headers for the `NEWS_API_KEY`. The question is, could we use this DIO `ApiClient`s other than the `NewsApiClient`? Perhaps no, because of that header not being required for others. 

With that thought in place, it is possible to imagine an scenario where we have `createNewsApiClientDio()` and `createOtherApiClientDio()`. 

Now the question is, how can we distinguish which one is used when creating the NewsApiClient?

Reply:
The same applies when we create multiple singletons of the same type, how does `get-it` distinguish?